language/move-prover/doc/user/spec-lang.md

```diff
@@ -206,6 +212,67 @@ The choice also comes in a form to select the *minimal* value from a set of inte
 choose min i: num where in_range(v, i) && v[i] == 2
 ```
 
+## Cast Operator
+
+In the specification language, we can use the same syntax `(e as T)` to cast an expression `e` with one integer type to
+`T`, an integer type of another size.
+
+
+## Shift Operator
+
+Shift operators `<<` and `>>` are supported in the specification language and both of them have the same semantics (including abort) with the Move language.
+
+## Bitwise Operators
+
+Move programs using bitwise operators `&`, `|` and `^` can be verified in the prover and these operators are also supported in the specification language.
+Due to encoding and efficiency issues, using bitwise operators has more caveats:
+
+- Integers involved in bitwise operations are encoded as `bv` types at the backend and two encodings of integers are not compatible. For instance,
+if a variable `v` involves in a bitwise operation such as `v & 2` or `v = a ^ b`, then when it is used in an arithmetic operation `v * w` or a shift operation `v << w`, `w` will be implicitly cast to a `bv` type in the Move program.
+However, the specification language does not support implicit type cast so users must explicitly use the builtin function `int2bv` in the specification: `v << int2bv(w)`.
+Not that since each `bv` type has a fixed length (from 8 to 256), values with type `num` cannot be converted into `bv`.
+
+
+- Verification of `bv` types is not efficient and may lead to timeout. As a result, users may prefer isolating bitwise operations from other operations and not using `int2bv`
+  if possible. Moreover, users need to use pragmas to explicitly specify which integer-typed function arguments or struct fields will be used in bitwise computations:
+
+```move
+    struct C has drop {
+        a: u64,
+        b: u64
+    }
+    spec C {
+        // b, the second field of C, will be of bv type
+        pragma bv=b"1";
+    }
+    public fun foo_generic<T>(i: T): T {
+      i
+    }
+
+    spec foo_generic {
+     // The first parameter will be of bv type if T is instantiated as a number type
+      pragma bv=b"0";
+     // The first return value will be of bv type if T is instantiated as a number type
+      pragma bv_ret=b"0";
+    }
+
+    public fun test(i: C): u64 {
+      let x1 = foo_generic(C.b);
+      x1 ^ x1
+    }
+
+    spec test {
+      // Explicit type cast is mandatory for generating correct boogie program
+      ensures result == (0 as u64);
+    }
+```
+
+Note that if arguments or fields of a generic function or struct are specified with `bv` types,
+they will be of `bv` types in all instances of the function or the struct.
```

Comment:
How about `bool` or `address`?

Reply:
`bool` or `address` will not be `bv` types.

Reply:
> bool or address will not be bv types.

Could you mention it in the text?

Reply:
done